bitcoin/src/light/wallet.rs

```diff
@@ -0,0 +1,639 @@
+use bitcoincore_rpc::bitcoin::{blockdata::constants::WITNESS_SCALE_FACTOR, PublicKey};
```

Comment:
There is a lot of complicated logic in this file. Did you manually translate all of it from the bitcoin core source? Are there no existing libraries we can use?

Reply:
Once this is in a reasonable state (see [suggested improvements](https://github.com/interlay/interbtc-clients/issues/369#issue-1372690752)) I think we can even contribute upstream.